# Problems Jupyter Notebook

## Problem 1: Extending the Lady Tasting Tea
![The Lady Tasting Tea](https://upload.wikimedia.org/wikipedia/en/2/2d/The_Lady_Tasting_Tea_-_David_Salsburg.jpg)

### Objective: 
_Let's extend the Lady Tasting Tea experiment as follows. The original experiment has 8 cups: 4 tea-first and 4 milk-first. Suppose we prepare 12 cups: 8 tea-first and 4 milk-first. A participant claims they can tell which was poured first._

_Simulate this experiment using `numpy` by randomly shuffling the cups many times and calculating the probability of the participant correctly identifying all cups by chance. Compare your result with the original 8-cup experiment._

_In your notebook, explain your simulation process clearly, report and interpret the estimated probability, and discuss whether, based on this probability, you would consider extending or relaxing the p-value threshold compared to the original design._

In [9]:
# Import libraries that will be used:
import math
import itertools
import numpy as np
import matplotlib.pyplot as plt

In [10]:
# Set up out our variables:
# Total number of cups
num_cups = 12
# Number of tea first
num_tea = 8
# Number of milk first
num_milk = 4

Using math.comb(n, k) to calculate combinations.
Interstingly, as we have by guessing the number of milk first cups correctly, we automatically know the number of tea first cups correctly as well. Therefore we only need to calculate one side of the combination. It would not make sense to select the number of tea first cups, however we will explore the alternative to see if it makes a difference.


In [11]:
# math.comb(n, k) calculates the number of combinations of n items taken k at a time.
ways = math.comb(num_cups, num_milk)
ways


495

Therefore above one can see that the total number of ways to correctly guess which cups had milk added first is 495. An increase from the original experiment from Fisher of 70.
Using a labelling and placeholder system comparing to the original problem we see that the number of placeholders does not change just the number of labelled cups:

Original problem (8 cups of tea 4 milk first 4 tea first)
Cup labels: `1` `2` `3` `4` `5` `6` `7` `8`
Placeholders: `_` `_` `_` `_`

New problem (12 cups of tea 4 milk first 8 tea first)
Cup labels: `1` `2` `3` `4` `5` `6` `7` `8` `9` `10` `11` `12`
Placeholders: `_` `_` `_` `_`

In [12]:
# If the order of the cups mattered it would result in:
order_ways = 12*11*10*9
order_ways

11880

We can see in the above that as we increase the number of cups if selecting from 12 cups in 4 places and the order is critical that there is a 1/11880 chance of it being correct. However in this situation we do not care about the order therefore we can disregard ordering.

In [ ]:
# Ways to shuffle our placeholders:
shuffles = 4*3*2*1
shuffles


24

In [ ]:
# Therfore the number of combinations is:
num_combs= order_ways // shuffles
num_combs

495

Therefore one can see that the num_combs = ways

## Problem 2: Normal Distribution

## Problem 3: t-Tests

## Problem 4: ANOVA

### References:
- A very helpful github guide on `markdown` used throughout for all markdown formatting [here](https://github.com/adam-p/markdown-here/wiki/markdown-cheatsheet)

## END